python/ray/air/tests/test_data_batch_conversion.py

```diff
@@ -296,4 +296,4 @@ def test_arrow_tensor_pandas(cast_tensor_columns):
 if __name__ == "__main__":
     import sys
 
-    sys.exit(pytest.main(["-sv", __file__]))
+    sys.exit(pytest.main(["-sv", __file__] + sys.argv[1:]))
```

Comment:
overall I think the direction is good. suggestions:

1. consider write a test macro (like `ray_py_test`) that just replaces the main entrance for the tests, and replace the tests with the macro
2. for these kind of large scale changes, maybe do the change incrementally, directory by directory, in multiple PR's, rather than trying to change all test setup in one PR at the same time. you can start with ones that you urgently need right now.